### PR TITLE
cmd: add `snap debug boot-vars` that dumps the current bootvars

### PR DESCRIPTION
**Pull request by mvo5**
The new debug helper `snap debug bootvars` will output the
current bootloader variables. This is useful on core18 systems
that do not have the grub-editenv/fw_getenv commands and for
bootloaders like little-kernel that do not have any tools for
this.

This will primarely be used in tests like:
https://github.com/snapcore/snapd/pull/7508

Or the test in
https://github.com/snapcore/snapd/pull/7438

### CODE REVIEW

File: boot/boot.go [hunk 1]
```diff
@@ -169,6 +170,23 @@ type NameAndRevision struct {
 	Revision snap.Revision
 }
 
+// DebugDumpBootvars writes a dump of the bootvars to the given writer
+func DebugDumpBootvars(w io.Writer) error {
+	bloader, err := bootloader.Find("", nil)
+	if err != nil {
+		return err
+	}
+	allKeys := []string{"snap_mode", "snap_core", "snap_try_core", "snap_kernel", "snap_try_kernel"}
```
Comment (bboozzoo) [inline boot/boot.go]: lk also has `reboot_reason` and `bootimg_file_name` though both seem to be an internal implementation detail.

Is `snap_menuentry` worth displaying here too?

File: cmd/snap/cmd_debug_bootvars.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/boot"
+)
+
+type cmdBootvars struct{}
+
+func init() {
+	cmd := addDebugCommand("bootvars",
+		"(internal) obtain the bootloader variables",
```
Comment (bboozzoo) [inline cmd/snap/cmd_debug_bootvars.go]: This should probably say `(internal) obtaind snapd boot variables` as we do not dump all boot variables

File: cmd/snap/cmd_debug_bootvars.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import (
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/boot"
+)
+
+type cmdBootvars struct{}
+
+func init() {
+	cmd := addDebugCommand("bootvars",
```
Comment (bboozzoo) [inline cmd/snap/cmd_debug_bootvars.go]: Should we also have `snap debug bootloaderenv` to be able to dump all the boot env tuples on core18 where there's no grub-editenv nor fw_printenv?

Comment (pedronis) [reply cmd/snap/cmd_debug_bootvars.go]: you mean not just ours?  we would need different helpers in the bootloader, seems a follow a up

Comment (pedronis) [reply cmd/snap/cmd_debug_bootvars.go]: I have a mild preference to call this boot-vars

File: boot/boot.go [hunk 4]
```diff
@@ -169,6 +170,23 @@ type NameAndRevision struct {
 	Revision snap.Revision
 }
 
+// DebugDumpBootvars writes a dump of the bootvars to the given writer
+func DebugDumpBootvars(w io.Writer) error {
```
Comment (pedronis) [inline boot/boot.go]: don't think it needs a "Debug" prefix, OTOH I would put it a debug.go file by itself

File: boot/debug.go [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/snapcore/snapd/bootloader"
+)
+
+// DumpBootvars writes a dump of the snapd bootvars to the given writer
+func DumpBootvars(w io.Writer) error {
```
Comment (pedronis) [inline boot/debug.go]: I missed that this is misspelled vs the rest of the package.  DumpBootVars  as we have GetBootVars etc ?

File: cmd/snap/cmd_debug_bootvars.go [hunk 6]
```diff
@@ -28,15 +28,15 @@ import (
 type cmdBootvars struct{}
 
 func init() {
-	cmd := addDebugCommand("bootvars",
-		"(internal) obtain the bootloader variables",
-		"(internal) obtain the bootloader variables",
+	cmd := addDebugCommand("boot-vars",
+		"(internal) obtaind snapd boot variables",
+		"(internal) obtaind snapd boot variables",
```
Comment (pedronis) [inline cmd/snap/cmd_debug_bootvars.go]: typo (it was in the PR comment):  s/obtaind/obtain/

**Pull request merged**